AR-Top-MobileApp/Assets/Prefabs/MapController.cs

```diff
@@ -234,41 +239,69 @@ void buildPiece (GridModel obj)
 				tilePiece.gameObject.SetActive (true);
 				tilePiece.transform.SetParent (modelLayer.transform);
 				colorize (tilePiece, obj.color);
+				rayCastText = tilePiece.GetComponent <RaycastText> ();
+				rayCastText.clearRaycastInfoText ();
+				rayCastInfo = "(" + obj.position.x.ToString () + ", " + obj.position.z.ToString () + ", " + (obj.position.y + 1).ToString () + ")";
+				rayCastText.appendRaycastInfoText (rayCastInfo);
 			}
 			break;
 		case "wall":
 			tilePiece = Instantiate (wallPrefab, tileVector, Quaternion.identity);
 			tilePiece.transform.SetParent (modelLayer.transform);
 			colorize (tilePiece, obj.color);
+			rayCastText = tilePiece.GetComponent <RaycastText> ();
+			rayCastText.clearRaycastInfoText ();
+			rayCastInfo = "(" + obj.position.x.ToString () + ", " + obj.position.z.ToString () + ", " + (obj.position.y + 1).ToString () + ")";
+			rayCastText.appendRaycastInfoText (rayCastInfo);
 			break;
 		case "floor":
 			tilePiece = Instantiate (floorPrefab, tileVector, Quaternion.identity);
 			tilePiece.transform.SetParent (modelLayer.transform);
 			colorize (tilePiece, obj.color);
+			rayCastText = tilePiece.GetComponent <RaycastText> ();
+			rayCastText.clearRaycastInfoText ();
+			rayCastInfo = "(" + obj.position.x.ToString () + ", " + obj.position.z.ToString () + ", " + (obj.position.y + 1).ToString () + ")";
+			rayCastText.appendRaycastInfoText (rayCastInfo);
 			break;
 		case "fighter":
 			tilePiece = Instantiate (fighterPrefab, tileVector, Quaternion.identity);
 			tilePiece.transform.SetParent (baseLayer.transform);
 			basePiece = tilePiece.transform.GetChild (1);
 			colorize (basePiece, obj.color);
+			rayCastText = tilePiece.GetComponent <RaycastText> ();
+			rayCastText.clearRaycastInfoText ();
+			rayCastInfo = "(" + obj.position.x.ToString () + ", " + obj.position.z.ToString () + ", " + (obj.position.y + 1).ToString () + ")";
+			rayCastText.appendRaycastInfoText (rayCastInfo);
```

Comment:
Can these four lines of code be moved outside the switch cases statement, since they are just duplicates of one another. It would make changing them much easier.

Reply:
I tried putting them outside the switch case statement but the tilePiece object doesn't get assigned if I do that. I can move them to a method, though.

Reply:
I would say no to that